### DISCUSSION TITLE
WinRE with wimboot - STATUS_WAIT_2 (0x80070002)

### DISCUSSION THREAD

**Discussion by gigolo91** [ipxe/ipxe#796] [Q&A] [answered]
Hi,

I try to boot with wimboot into WinRE (Recovery Environment). It looks like that WinRE is booting normally. 
But when I try to start "System Image Recovery" I get the following error:

```
An internal error occurred. The following information might help you resolve the error:
STATUS_WAIT_2 (0x80070002)
```

I guess that winre.wim is ok. If I boot into Winre.wim with our WDS setup, I can start the System Image Recovery normally, without error. 

The ipxe setup looks like this:

```
:winre
set winre http://192.168.1.1/winRE
kernel ${winre}/wimboot pause
initrd ${winre}/boot/BCD                     BCD
initrd ${winre}/boot/boot.sdi                boot.sdi
initrd -n boot.wim ${base-url}/Winre.wim        boot.wim
boot
```

Is anybody able to boot into WinRE with wimboot and use System Image Recovery? Any hints, why System Image Recovery is not possible?

Thx!!

**Comment by NiKiZe** [comment]
Have you checked if the image has multiple indexes?
Maybe one is bootable, the other is not?

Does this error show in bootmgr or is it inside winpe?
Any chance of a screenshot?

It might be other options you need in BCD. But I would drop both boot.sdi and BCD from your script since those are extracted from wim when possible.

**Reply by gigolo91** [reply]
Thx for the quick response!
Only 1 index, the wim is bootable, I get the error in WinRE, when I hit System Image Recovery.

I dropped boot.sdi and BCD from iPxe, my setup now looks like:

```
:winre
set winre http://192.168.1.1/winRE
kernel ${winre}/wimboot
initrd -n boot.wim ${base-url}/Winre.wim        boot.wim
boot
```

But still no success. Here are some screenshots:

![image](https://user-images.githubusercontent.com/78353320/200559946-e68c2694-71ef-4743-8a40-1aaeda579c6a.png)
![image](https://user-images.githubusercontent.com/78353320/200560001-2972b9d3-82ed-41fd-9f0d-0ec548fc27f9.png)
![image](https://user-images.githubusercontent.com/78353320/200560029-846e831d-8fed-46d6-a182-7ceb5cbba477.png)

Boot per WDS over tftp, same Winre.wim. Takes forever, but it is working:

![image](https://user-images.githubusercontent.com/78353320/200560450-623231ba-6313-4b3a-a1ed-a3dcc6df2dfd.png)

**Reply by stappersg** [reply]
Not an `iPXE` issue, maybe a [`wimboot`](https://github.com/ipxe/wimboot) issue.

**Reply by NiKiZe** [reply]
I think settings are provided to the imaging software.
This is more about how the backup works rather then wimboot. There is several issues on the net. Not saying there is any good documented fixes. But I think you will have to debug this further and try to narrow down the cause.

Where should it find the image that is mentioned?

**Reply by gigolo91** [reply]
I exported the Winre.wim out of the install.wim out of an Win10-iso. 
So mount iso -> mount install.wim with dism -> and then Windows\System32\Recovery

What I don't understand is that the winre.wim works, if I boot into the winre.wim with WDS/TFTP. So it looks like that wimboot is handling something different.

**Reply by mcb30** [reply]
WDS will provide additional information about the TFTP server, which the Windows recovery process is almost certainly picking up and using to locate the actual recovery image (as opposed to the boot image).

This information is not available when you are booting your `winre.wim` directly via wimboot.  You will instead need to provide an alternative way for the boot image to locate the recovery image.  See e.g. https://ipxe.org/howto/winpe#installing_windows - the documentation here talks about locating an installation image (rather than a recovery image), but the principle will be the same.

**Comment by gigolo91** [comment, marked as answer]
I found out that System Image Recovery is calling bmrui.exe.
With that info I landed in the ipxe forum:
https://forum.ipxe.org/printthread.php?tid=8137

If I regedit
"SystemStartOptions"=" MININT REDIRECT RDIMAGEOFFSET=8192 RDIMAGELENGTH=3161088 RDPATH=MULTI(0)DISK(0)RDISK(0)PARTITION(1)\SOURCES\BOOT.WIM"
to
"SystemStartOptions"=" MININT REDIRECT RDIMAGEOFFSET=8192 RDIMAGELENGTH=3161088 RDPATH=MULTI(0)DISK(0)CDROM(0)\SOURCES\BOOT.WIM"
I can start System Image Recovery without errors. :-)

Registry key can be found at: HKLM\System\CurrentControlSet\Control
